### PR TITLE
fix: ping return bulk string

### LINKED ISSUE #926: PING returns PONG if no argument is provided, otherwise return a copy of the argument as a bulk.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Version

unstable

### Minimal reproduce step

ping message

### What did you expect to see?

message

### What did you see instead?

wrong number of arguments

### Anything Else?

[redis ping](https://redis.io/commands/ping/)

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by ColinChamber**
This closes #926.

### CODE REVIEW

File: tests/gocase/unit/introspection/introspection_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (PragmaTwice) [inline tests/gocase/unit/introspection/introspection_test.go:1]: May I ask the meaning of the filename? I think something like `ping_test.go` will be more simple.

Comment (ColinChamber) [reply tests/gocase/unit/introspection/introspection_test.go:1]: I did this to imitate [redis](https://github.com/redis/redis/blob/unstable/tests/unit/introspection.tcl#L3-L5). Kvrocks also has introspection.tcl

File: src/redis_cmd.cc [hunk 2]
```diff
@@ -242,7 +242,13 @@ class CommandFlushAll : public Commander {
 class CommandPing : public Commander {
  public:
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
-    *output = Redis::SimpleString("PONG");
+    if (args_.size() == 1) {
+      *output = Redis::SimpleString("PONG");
+    } else if (args_.size() == 2) {
+      *output = Redis::SimpleString(args_[1]);
```
Comment (tisonkun) [inline src/redis_cmd.cc]: IIRC Redis will return bulk string here. That is,

```bash
$ echo -e '*1\r\n$4\r\nPING\r\n' | nc localhost 6379
+PONG

$ echo -e '*2\r\n$4\r\nPING\r\n$4\r\nPONG\r\n' | nc localhost 6379
$4
PONG
```

File: tests/gocase/unit/introspection/introspection_test.go (hunk 1 above)
Comment (PragmaTwice) [reply tests/gocase/unit/introspection/introspection_test.go:1]: Oh, got it. Thanks.

File: src/redis_cmd.cc (hunk 2 above)
Comment (caipengbo) [reply src/redis_cmd.cc]: How do we determine simple string and bulk string in the go test case? @tisonkun

Comment (tisonkun) [reply src/redis_cmd.cc]: Using `server.NewTCPClient` to obtain a low-level client to verify the raw response.

Comment (caipengbo) [reply src/redis_cmd.cc]: Ok, thank you. That's exactly what I need :)

Comment (ColinChamber) [reply src/redis_cmd.cc]: Thank you. This is my mistake. My branch name is ping-return-bulk-string, but I still misuse the simple string. 😓

File: tests/gocase/unit/introspection/introspection_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	rdb := srv.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+
+	t.Run("PING", func(t *testing.T) {
+		c := srv.NewTCPClient()
+		defer func() { require.NoError(t, c.Close()) }()
+		require.NoError(t, c.Write("*1\r\n$4\r\nPING\r\n"))
+		r, err := c.ReadLine()
+		require.NoError(t, err)
+		require.Contains(t, r, "+PONG")
```
Comment (caipengbo) [inline tests/gocase/unit/introspection/introspection_test.go]: Should we write a new test case for the redis protocol validation about PING?

It seems that semantic verification and protocol verification have been put into one test case, which I think should be split into two cases.

File: tests/gocase/unit/introspection/introspection_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	srv := util.StartServer(t, map[string]string{})
+	defer srv.Close()
+	ctx := context.Background()
+	rdb := srv.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+
+	t.Run("PING", func(t *testing.T) {
+		c := srv.NewTCPClient()
+		defer func() { require.NoError(t, c.Close()) }()
+		require.NoError(t, c.Write("*1\r\n$4\r\nPING\r\n"))
```
Comment (tisonkun) [inline tests/gocase/unit/introspection/introspection_test.go]: use `WriteArgs`

File: tests/gocase/unit/introspection/introspection_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		r, err := c.ReadLine()
+		require.NoError(t, err)
+		require.Contains(t, r, "+PONG")
+		require.NoError(t, c.Write("*2\r\n$4\r\nPING\r\n$4\r\nPONG\r\n"))
```
Comment (tisonkun) [inline tests/gocase/unit/introspection/introspection_test.go]: use `WriteArgs`

File: tests/gocase/unit/introspection/introspection_test.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+		r, err = c.ReadLine()
+		require.NoError(t, err)
+		require.Contains(t, r, "$4")
+		r, err = c.ReadLine()
+		require.NoError(t, err)
+		require.Contains(t, r, "PONG")
+		require.EqualError(t, rdb.Do(ctx, "ping", "hello", "redis").Err(), "ERR wrong number of arguments")
```
Comment (tisonkun) [inline tests/gocase/unit/introspection/introspection_test.go]: consistently use `TCPClient` and just remove `rdb` outside.

**Review by tisonkun — COMMENTED** [review]
Generally looks good. Comments inline.

File: tests/gocase/unit/introspection/introspection_test.go (hunk 3 above)
Comment (ColinChamber) [reply tests/gocase/unit/introspection/introspection_test.go]: Yeah. Maybe it is not necessary to verify the protocol about PING. Some things are apparent. It makes the test too cumbersome.

File: tests/gocase/unit/introspection/introspection_test.go [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+	srv := util.StartServer(t, map[string]string{})
+	defer srv.Close()
+
+	t.Run("PING", func(t *testing.T) {
+		c := srv.NewTCPClient()
+		defer func() { require.NoError(t, c.Close()) }()
+		require.NoError(t, c.WriteArgs("PING"))
+		r, err := c.ReadLine()
+		require.NoError(t, err)
+		require.Contains(t, r, "+PONG")
```
Comment (tisonkun) [inline tests/gocase/unit/introspection/introspection_test.go]: I think you should use `require.Equal` here? Also below assertions seem looser than it should be.

Comment (ColinChamber) [reply tests/gocase/unit/introspection/introspection_test.go]: Thank you. Updated.

**Review by tisonkun — APPROVED** [review]

File: src/redis_cmd.cc [hunk 8]
```diff
@@ -5833,7 +5839,7 @@ class CommandXTrim : public Commander {
 
 CommandAttributes redisCommandTable[] = {
     ADD_CMD("auth", 2, "read-only ok-loading", 0, 0, 0, CommandAuth),
-    ADD_CMD("ping", 1, "read-only", 0, 0, 0, CommandPing),
+    ADD_CMD("ping", -1, "read-only", 0, 0, 0, CommandPing),
```
Comment (tisonkun) [inline src/redis_cmd.cc:5842]: May I ask the reason to make this change? Is there some reference from the Redis document?

The overall change looks good to me. Merging...

Comment (ColinChamber) [reply src/redis_cmd.cc:5842]: This digit indicates the number of parameters allowed(including command name), "1 " means there is precisely one, and "-1 "means at least one.  Do you want this [source code](https://github.com/redis/redis/blob/6.2/src/server.c#L715) and the [comment](https://github.com/redis/redis/blob/6.2/src/server.c#L89)?

Comment (tisonkun) [reply src/redis_cmd.cc:5842]: Thank you!

**Pull request merged**